### PR TITLE
Add shader-compiler option and versions 1.0.6, 1.0.7 and 1.0.8

### PR DESCRIPTION
**Pull request by psyinf**
**vsg/1.0.6-1.0.8**

Depends on #19980

updated versions and re-enabled shader-compiler option starting with 1.0.5

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/vsg/all/conandata.yml [hunk 1]
```diff
@@ -8,3 +8,12 @@ sources:
   "1.0.5":
     url: "https://github.com/vsg-dev/VulkanSceneGraph/archive/refs/tags/v1.0.5.tar.gz"
     sha256: "ff58260fcb88d19d92c40a70bc40ff06abb1a8805568eb76862a036d13ada75b"
+  "1.0.6":
```
Comment (uilianries) [inline recipes/vsg/all/conandata.yml:11]: These versions 1.0.6 and 1.0.7 are really required? Could you update only to 1.0.8? We kind of avoid updating to outdated versions, as most are unused.

Comment (psyinf) [reply recipes/vsg/all/conandata.yml:11]: You're probably right. I'll remove them.

**Review by uilianries — CHANGES_REQUESTED** [review]
Thank you for your contribution. I'm concerned about the `shader_compiler` option, as it will install another project by git command.

File: recipes/vsg/all/conanfile.py [hunk 2]
```diff
@@ -11,20 +11,26 @@
...
     license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.vulkanscenegraph.org"
-    topics = ("vulkan", "scenegraph", "graphics", "3d")
+    topics = ("vulkan", "scenegraph", "graphics", "3d", "VulkanSceneGraph")
     package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
+        "shader_compiler": [True, False],
```
[the remark this answers is not in the document]
Comment (psyinf) [reply recipes/vsg/all/conanfile.py:27]: Without the shader_compiler the VSG is not usable to its full extent. The original author integrates glslang here by adding directly into the source graph of VSG and uses headers that are not exposed by the glslang conan-package. I've tried really hard to integrate a custom glslang-package, but this required massive changes in the source-structure which are not maintainable on the long-run. As glslang-sources are inlined with VSG and the author likes to control which exact versions to be bundled with VSG I don't really get the argument about control over version. 

I also tried to convince the author not to use git-submodules or direct checkouts but he arguments that he wants to have VSG as self-contained as possible. I can start another attempt to create a custom vsg-glslang package, but honestly I don't think I will have the time in the long run to maintain changes here by creating patches. Still, I'll try. 

Sorry for inconveniences, but I thought checking it out into the own source-graph should be ok, as it is not visible from the outside, thus not interfering downstream with other packages.

File: recipes/vsg/all/conanfile.py [hunk 3]
```diff
@@ -11,20 +11,26 @@
...
+    description = "VulkanSceneGraph (VSG), is a modern, cross platform, \
+                   high performance scene graph library \
+                   built upon Vulkan graphics/compute API. \
+                   The software is written in C++17, \
+                   and follows the CppCoreGuidelines and FOSS Best Practices."
     license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.vulkanscenegraph.org"
-    topics = ("vulkan", "scenegraph", "graphics", "3d")
+    topics = ("vulkan", "scenegraph", "graphics", "3d", "VulkanSceneGraph")
```
[the remark this answers is not in the document]
Comment (psyinf) [reply recipes/vsg/all/conanfile.py:22]: Curiously this leads to a package that cannot be found in the conan-center web interface by looking for "VulkanSceneGraph", which is the official name. Any clues here?

File: recipes/vsg/all/conanfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (psyinf) [reply recipes/vsg/all/conanfile.py:27]: I revisited the issue and found the culprit being a wrong assumption about the glslang version used. I was able to tackle the issue by adding a new version to glslang. So basically this PR has to wait for the glslang-PR completed. (link TBD)

**Pull request closed without merging**